pkgs/by-name/tp/tplay/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  pkg-config,
+  openssl,
+  opencv,
+  clang,
+  libclang,
+  ffmpeg,
+  alsa-lib,
+  fetchFromGitHub,
+  rustPlatform
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "tplay";
+  version = "0.45";
+
+  src = fetchFromGitHub {
+    owner = "maxcurzi";
+    repo = pname;
+    rev =  "v${version}";
+    hash = "sha256-qt5I5rel88NWJZ6dYLCp063PfVmGTzkUUKgF3JkhLQk=";
+  };
+
+  cargoHash = "sha256-jgUKGV2Yg8+iF2wQZd1Z+QFfyJmywVTFVECUs+TK8zA=";
+  cargoPatches = [ ./cargo.diff ];
```

Comment:
Any explanations for the 2000+ lines patch?

Reply:
this is a cargo.lock, i think to import this but i make it works